src/switch_ivr_bridge.c

```diff
@@ -474,7 +474,9 @@ static void *audio_bridge_thread(switch_thread_t *thread, void *obj)
 	user_data = data->session_data;
 	stream_id = data->stream_id;
 
-	per_read_frame_ms = 1000 / (read_impl.actual_samples_per_second / read_impl.samples_per_packet);
+	if (read_impl.actual_samples_per_second != 0 && read_impl.samples_per_packet != 0) {
+		per_read_frame_ms = 1000 / (read_impl.actual_samples_per_second / read_impl.samples_per_packet);
```

Comment:
Is it better to just check if `switch_core_session_get_read_impl` return a success? Also if we are unable to get a codec for read not sure if we will need to proceed with the bridge as this might cause another issue since read_impl will be use below without any validation.

Reply:
Another is to investigate the reason why it proceeded with the bridge even if there's no read codec implementation. I think it should have been detected also it doesn't make sense to bridge an audio if we don't have yet the read codec implementation.

Reply:
that made me wonder too... could this also be caused by an overwrite of the session object? `read_impl` belongs to the session object..

Reply:
Might be possibility but will need to have the evidence of it happening. Or might be a loop hole that might have skip the validation. Is this crash happening recently or we have this issue before 1.10.7?

Reply:
this is a new one, I checked that with @minhtuan1407-telnyx

Reply:
Check for the return code of `switch_core_session_get_read_impl` has been added.